lib/HTFeed/BackupExpiration.pm

```diff
@@ -31,22 +31,26 @@ sub run {
 
   my $config = get_config('storage_classes');
   my $storage_config = $config->{$self->{storage_name}};
-  # Get a list of all volumes with this storage name that have one or more old
-  # versions. This query is about finding sets of volumes, nothing more granular.
+  # Get a list of all volumes with this storage name that have at least one new version
+  # and at least one old version, and at least three in all. Those that have multiple old
+  # versions will be culled below.
   my $sth = get_dbh()->prepare(<<'SQL');
-    SELECT namespace,id,MAX(version) AS max_version
+    SELECT namespace,id
```

Comment:
I think we just need to find everything with more than one version that's more than 6 months old, regardless of any versions < 6 months old (which we never touch):

```sql
SELECT namespace, id
FROM feed_backups
WHERE deleted IS NULL
    AND storage_name=?
    AND version < DATE_FORMAT(DATE_SUB(NOW(), INTERVAL 180 DAY),"%Y%m%d%H%i%S")
    GROUP BY namespace,id
    HAVING count(*) >= 2
```